Type hints: add `py.typed` marker

## Description of the problem, including code/CLI snippet

```
(.venv) C:\Build\project>echo import gitlab > bug.py

(.venv) C:\Build\project>mypy bug.py
```

## Expected Behavior

No `mypy` error

## Actual Behavior

```
bug.py:1: error: Skipping analyzing "gitlab": module is installed, but missing library stubs or py.typed marker  [import-untyped]
bug.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
bug.py:1: note: See https://mypy.rtfd.io/en/stable/_refs.html#code-import-untyped for more info
Found 1 error in 1 file (checked 1 source file)
```

## Specifications

  - python-gitlab version: 4.1.0 and 4.0.0 (not before 4)
  - API version you are using (v3/v4): n/a
  - Gitlab server version (or gitlab.com): n/a

## Relevant info

> Mypy will not try inferring the types of any 3rd party libraries you have installed unless they either have declared themselves to be [PEP 561 compliant stub package](https://mypy.readthedocs.io/en/stable/installed_packages.html#installed-packages) (e.g. with a py.typed file) or have registered themselves on [typeshed](https://github.com/python/typeshed), the repository of types for the standard library and some 3rd party libraries.

https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-library-stubs-or-py-typed-marker

And in fact, `echo > .venv\Lib\site-packages\gitlab\py.typed` fixes the problem for me.

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -20,6 +20,7 @@ jobs:
         token: ${{ secrets.RELEASE_GITHUB_TOKEN }}
 
     - name: Python Semantic Release
+      id: release
       uses: python-semantic-release/python-semantic-release@v8.3.0
       with:
         github_token: ${{ secrets.RELEASE_GITHUB_TOKEN }}

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,4 +1,4 @@
 include COPYING AUTHORS CHANGELOG.md requirements*.txt
-include tox.ini
+include tox.ini gitlab/py.typed
 recursive-include tests *
 recursive-include docs *j2 *.js *.md *.py *.rst api/*.rst Makefile make.bat

--- a/tests/smoke/test_dists.py
+++ b/tests/smoke/test_dists.py
@@ -12,6 +12,7 @@ DOCS_DIR = "docs"
 TEST_DIR = "tests"
 SDIST_FILE = f"{__title__}-{__version__}.tar.gz"
 WHEEL_FILE = f"{__title__.replace('-', '_')}-{__version__}-py{sys.version_info.major}-none-any.whl"
+PY_TYPED = "gitlab/py.typed"
 
 
 @pytest.fixture(scope="session")
@@ -21,17 +22,24 @@ def build(tmp_path_factory: pytest.TempPathFactory):
     return temp_dir
 
 
-def test_sdist_includes_docs_and_tests(build: Path) -> None:
+def test_sdist_includes_correct_files(build: Path) -> None:
     sdist = tarfile.open(build / SDIST_FILE, "r:gz")
     sdist_dir = f"{__title__}-{__version__}"
 
     docs_dir = sdist.getmember(f"{sdist_dir}/{DOCS_DIR}")
     test_dir = sdist.getmember(f"{sdist_dir}/{TEST_DIR}")
     readme = sdist.getmember(f"{sdist_dir}/README.rst")
+    py_typed = sdist.getmember(f"{sdist_dir}/{PY_TYPED}")
 
     assert docs_dir.isdir()
     assert test_dir.isdir()
+    assert py_typed.isfile()
     assert readme.isfile()
+
+
+def test_wheel_includes_correct_files(build: Path) -> None:
+    wheel = zipfile.ZipFile(build / WHEEL_FILE)
+    assert PY_TYPED in wheel.namelist()
 
 
 def test_wheel_excludes_docs_and_tests(build: Path) -> None:

